frontend/src/components/BackupCreateDialog.vue

```diff
@@ -0,0 +1,329 @@
+import Notification from '@/models/notification';
+import { usePrincipalStore, useNotificationsStore } from '@/stores';
+import { ref, watch } from 'vue'
+import PricePrediction from '@/components/PricePrediction.vue';
+import ComplianceCheck from '@/components/ComplianceCheck.vue';
+
+
+const principalStore = usePrincipalStore();
+const notificationsStore = useNotificationsStore();
+
+const model = defineModel<boolean>();
+
+const isLoading = ref(true);
+const sourceProjects = ref<string[]>([]);
+const storageClasses = ref<{title: string, value: string}[]>([]);
+const storageRegions = ref<string[]>([]);
+const backupTypes = ref([{title: 'Cloud Storage', value: 'CloudStorage'}, {title: 'BigQuery', value: 'BigQuery'}]);
+const strategies = ref(['Snapshot', 'Oneshot', 'Mirror']);
+
+const loadingBucketNames = ref(false);
+const bucketNames = ref<string[]>([]);
+const loadingDatasetNames = ref(false);
+const datasetNames = ref<string[]>([]);
+
+const request = ref<CreateRequest>({
+    gcs_options: {},
+    bigquery_options: {},
+    target: {},
+    snapshot_options: {},
+})
+const isValid = ref(false);
+
+const evalutingBackup = ref<CreateRequest|undefined>();
+
+const updateData = async () => {
+    request.value = {
+        gcs_options: {},
+        bigquery_options: {},
+        target: {},
+        snapshot_options: {},
+    };
+    evalutingBackup.value = undefined;
+    isLoading.value = true;
+    sourceProjects.value = principalStore.principal.getProjects()
+    Promise.all([
+        DefaultService.getConfigRegions(),
+        DefaultService.getConfigStorageClasses(),
+    ]).then(([regionResponse, storageClassResponse]) => {
+        storageClasses.value = storageClassResponse.storage_classes?.map((c) => { return {
+            title: capitalize(c.toLowerCase()),
+            value: c,
+        }}) ?? [];
+        storageRegions.value = regionResponse.regions ?? [];
+    }).catch((err) => notificationsStore.handleError(err)).finally(() => {
+        isLoading.value = false;
+    });
+}
+
+const updateBucketNames = () => {
+    bucketNames.value = [];
+    if(request.value.project) {
+        loadingBucketNames.value = true;
+        DefaultService.getBuckets(request.value.project).then((response) => {
+            bucketNames.value = response.buckets ?? [];
+        }).catch((err) => notificationsStore.handleError(err))
+        .finally(() => {
+            loadingBucketNames.value = false;
+        })
+    }
+}
+
+const updateDatasetNames = () => {
+    bucketNames.value = [];
+    if(request.value.project) {
+        loadingDatasetNames.value = true;
+        DefaultService.getDatasets(request.value.project).then((response) => {
+            datasetNames.value = response.datasets ?? [];
+        }).catch((err) => notificationsStore.handleError(err))
+        .finally(() => {
+            loadingDatasetNames.value = false;
+        })
+    }
+}
+
+const updateSourceFields = () => {
+    if(request.value.type == 'CloudStorage') {
```

Comment:
**suggestion**: replace values for backup type `CloudStorage` and `BigQuery` with const (from model?)